src/zkbob/ZkBobPool.sol

```diff
@@ -235,19 +237,28 @@ abstract contract ZkBobPool is IZkBobPool, EIP1967Admin, Ownable, Parameters, Zk
         int256 token_amount = transfer_token_delta + int256(fee);
         int256 energy_amount = _transfer_energy_amount();
 
+        require(
+            token_amount
+                == (token_amount * int256(TOKEN_DENOMINATOR) / int256(TOKEN_NUMERATOR)) / int256(TOKEN_DENOMINATOR)
+                    * int256(TOKEN_NUMERATOR),
+            "ZkBobPool: incorrect token amount"
+        );
```

Comment:
Seems like it's the same as:
```suggestion
        require(token_amount % int256(TOKEN_NUMERATOR) == 0, "ZkBobPool: incorrect token amount");
```
When numerator is 1, it does nothing, when denominator is 1 - it verifies that precision matches numerator.